src/simulation_lifecycle/utils/workflow.c

```diff
@@ -0,0 +1,99 @@
+#include <string.h>
+#include <unistd.h>
+#include "cJSON.h"
+#include "simulation_lifecycle/error.h"
+#include "simulation_lifecycle/utils/file.h"
+#include "simulation_lifecycle/utils/workflow.h"
+
+
+cJSON * read_workflow_file(char *path_to_file){
+    cJSON *content = NULL;
+    read_json_file(path_to_file, &content);
+    return(content);
+}
+
+cJSON * read_data_sources(cJSON * workflow){
+    return  cJSON_GetObjectItem(workflow, "data_sources");
+}
+
+cJSON * read_simulation(cJSON * workflow){
+    return  cJSON_GetObjectItem(workflow, "simulation");
+}
+
+cJSON * read_spatial_analysis(cJSON * workflow){
+    return  cJSON_GetObjectItem(workflow, "spatial_analysis");
+}
+
+cJSON * read_visualization(cJSON * workflow){
+    return  cJSON_GetObjectItem(workflow, "visualization");
+}
+
+int validate_workflow(cJSON * workflow){
+    //Check workflow
+    if(workflow == NULL){
+        return WORKFLOW_DOES_NOT_EXIST;
+    }
+
+    if(workflow->child == NULL){
+        return WORKFLOW_CONTAINS_EMPTY_JSON_OBJ;
+    }
```

Comment:
This works and is what I originally used. But my feeling now is that child is not meant to be accessed directly like this. I think in this case you could check cJSON_GetArraySize. Although it seems like it's meant specifically for arrays, it also works on objects, it returns the number of properties.